game-core/src/main/java/games/strategy/engine/framework/system/DevOverrides.java

```diff
@@ -0,0 +1,64 @@
+package games.strategy.engine.framework.system;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.logging.Level;
+import lombok.experimental.UtilityClass;
+import lombok.extern.java.Log;
+
+/**
+ * Utility class to read override values present on local file system. These overrides are not meant
+ * to be used by an end user but instead during development. If the file is present and if a
+ * property has a value set, then we will use it.
+ */
+@UtilityClass
+@Log
+public class DevOverrides {
+  private static final String OVERRIDE_FILE_NAME = "dev-overrides.properties";
+
+  public static Optional<URI> readMapServerOverride() {
+    final Properties properties = readPropertiesFromOverrideFile();
+
+    final String value = properties.getProperty("maps.server", "");
+    return value.isBlank() ? Optional.empty() : Optional.of(URI.create(value));
+  }
+
+  private Properties readPropertiesFromOverrideFile() {
+    final Properties properties = new Properties();
+    findOverrideFile()
+        .ifPresent(
+            overrideFile -> {
+              try (InputStream fileInput = Files.newInputStream(overrideFile)) {
+                properties.load(fileInput);
+              } catch (final IOException e) {
+                log.log(
+                    Level.SEVERE,
+                    "Failed to read override file: " + overrideFile.toFile().getAbsolutePath(),
```

Comment:
No need to convert it to `java.io.File` first
```suggestion
                    "Failed to read override file: " + overrideFile.toAbsolutePath(),
```